### PR TITLE
facedetect / opencv : fix build cv < 3.0

### PR DESCRIPTION
**Pull request by d-j-a-y**

### CODE REVIEW

**Comment by jaromil** [comment]
utACK, I believe this is needed and would merge even without testing with both versions. CI Tests pass. thanks!

**Comment by rrrapha** [comment]
Both versions work for me with OpenCV 2.4.13 on OpenBSD. 

..but wouldn't it be better to just include all the opencv headers like this?
`#include <opencv2/opencv.hpp>`

File: src/filter/facedetect/facedetect.cpp [hunk 1]
```diff
@@ -25,8 +25,19 @@
...
+#define CV_VERSION_NUM (CV_MAJOR_VERSION * 10000 \
+                      + CV_MINOR_VERSION * 100 \
+                      + CV_VERSION_REVISION)
+
+#if CV_VERSION_NUM > 30000
+  #include <opencv2/objdetect.hpp>
+  #include <opencv2/imgproc.hpp>
+#else
+  #include <opencv2/objdetect/objdetect.hpp>
+  #include <opencv2/imgproc/imgproc.hpp>
```
Comment (ddennedy) [inline src/filter/facedetect/facedetect.cpp]: The latter form works on OpenCV 3.2 as well.

**Comment by ddennedy** [comment]
> but wouldn't it be better to just include all the opencv headers like this?
> `#include <opencv2/opencv.hpp>`

That works for me as well. I suggested module-specific form because I seem to recall it was suggested, and it is generally better practice in modular projects (e.g. Qt). However, considering the difficulty we are facing with OpenCV's header changes, I think it is better to take the safer route. Can we please all test that general include? It works for me on OpenCV 3.2.

**Comment by d-j-a-y** [comment]
> ..but wouldn't it be better to just include all the opencv headers like this?
> #include <opencv2/opencv.hpp>

works for me (deb9 opencv 2.4.9.1)

**Pull request closed without merging**